py-rattler/tests/unit/test_version.py

```diff
@@ -17,3 +18,17 @@ def test_version_dash_normalisation() -> None:
 
     with pytest.raises(Exception):
         Version("1-.0dev+3.4-")
+
+
+def test_bump_fail() -> None:
+    with pytest.raises(VersionBumpError):
+        Version("1").bump_minor()
+
+    with pytest.raises(VersionBumpError):
+        Version("1.5").bump_patch()
+
+    with pytest.raises(VersionBumpError):
+        Version("1.5").bump_segment(-5)
+
+    with pytest.raises(VersionBumpError):
+        Version("1.5").bump_segment(5)
```

Comment:
Could you move these to the doctest of `bump_segment`? @Wackyator this is possible right?

Reply:
Should be good now.